src/test/ui/or-patterns/or-patterns-syntactic-fail.stderr

```diff
@@ -0,0 +1,112 @@
+   |
+LL |     fn fun(A | B: E) {}
+   |              ^ expected one of `:` or `@` here
+
+error: a leading `|` is only allowed in a top-level pattern
+  --> $DIR/or-patterns-syntactic-fail.rs:37:11
+   |
+LL |     let ( | A | B) = E::A;
+   |           ^ help: remove the `|`
+
+error: a leading `|` is only allowed in a top-level pattern
+  --> $DIR/or-patterns-syntactic-fail.rs:38:11
+   |
+LL |     let ( | A | B,) = (E::B,);
+   |           ^ help: remove the `|`
+
+error: a leading `|` is only allowed in a top-level pattern
+  --> $DIR/or-patterns-syntactic-fail.rs:39:11
+   |
+LL |     let [ | A | B ] = [E::A];
+   |           ^ help: remove the `|`
+
+error: a leading `|` is only allowed in a top-level pattern
+  --> $DIR/or-patterns-syntactic-fail.rs:40:13
+   |
+LL |     let TS( | A | B );
+   |             ^ help: remove the `|`
+
+error: a leading `|` is only allowed in a top-level pattern
+  --> $DIR/or-patterns-syntactic-fail.rs:41:17
+   |
+LL |     let NS { f: | A | B };
+   |                 ^ help: remove the `|`
+
+error: a leading `|` is only allowed in a top-level pattern
+  --> $DIR/or-patterns-syntactic-fail.rs:43:11
+   |
+LL |     let ( || A | B) = E::A;
+   |           ^^ help: remove the `||`
+
+error: a leading `|` is only allowed in a top-level pattern
+  --> $DIR/or-patterns-syntactic-fail.rs:44:11
+   |
+LL |     let [ || A | B ] = [E::A];
+   |           ^^ help: remove the `||`
+
+error: a leading `|` is only allowed in a top-level pattern
+  --> $DIR/or-patterns-syntactic-fail.rs:45:13
+   |
+LL |     let TS( || A | B );
+   |             ^^ help: remove the `||`
+
+error: a leading `|` is only allowed in a top-level pattern
+  --> $DIR/or-patterns-syntactic-fail.rs:46:17
+   |
+LL |     let NS { f: || A | B };
+   |                 ^^ help: remove the `||`
+
+error: no rules expected the token `|`
+  --> $DIR/or-patterns-syntactic-fail.rs:14:15
+   |
+LL | macro_rules! accept_pat {
+   | ----------------------- when calling this macro
+...
+LL | accept_pat!(p | q);
+   |               ^ no rules expected this token in macro call
+
+error: no rules expected the token `|`
+  --> $DIR/or-patterns-syntactic-fail.rs:15:13
+   |
+LL | macro_rules! accept_pat {
+   | ----------------------- when calling this macro
+...
+LL | accept_pat!(| p | q);
+   |             ^ no rules expected this token in macro call
+
+warning: the feature `or_patterns` is incomplete and may cause the compiler to crash
+  --> $DIR/or-patterns-syntactic-fail.rs:4:12
+   |
+LL | #![feature(or_patterns)]
+   |            ^^^^^^^^^^^
+   |
+   = note: `#[warn(incomplete_features)]` on by default
+
+error[E0369]: binary operation `|` cannot be applied to type `E`
+  --> $DIR/or-patterns-syntactic-fail.rs:24:22
+   |
+LL |     let _ = |A | B: E| ();
+   |                  ----^ -- ()
+   |                  |
+   |                  E
+   |
+   = note: an implementation of `std::ops::BitOr` might be missing for `E`
```

Comment:
HA. Here we have an example :)

Let's leave it as is but it is weeeeird.

Reply:
Yep. Note that this is primarily to check how the parser behaves per the specification in the RFC. In this case we actually do have an ambiguity because this is an expression context unlike the case above with the `f: || A | B` since that is a pattern context.